### PR TITLE
neovim: init

### LINKED ISSUE #471: neovim: extract from VIM module
The `stylix.targets.vim.enable` option guards

https://github.com/danth/stylix/blob/29148118cc33f08b71058e1cda7ca017f5300b51/modules/vim/hm.nix#L57-L58

For consistancy, we should add a `stylix.targets.neovim.enable` option and declare it in a new `modules/neovim/hm.nix` module.

Whether `stylix.targets.neovim.enable` and `stylix.targets.vim.enable` should be aliases is still an open question.

### PR DESCRIPTION
**Pull request by donovanglover**
Closes #471

### CODE REVIEW

**Review by trueNAHO — CHANGES_REQUESTED** [review]
I have successfully tested that this PR does not interfere with NixVim when Neovim is disabled.

Consider taking inspiration from the very similar [NixVim implementation](https://github.com/danth/stylix/blob/feb2973dfa8232c07efbd2b48f11a5cfa2276570/modules/nixvim/nixvim.nix).

**Review by trueNAHO — APPROVED** [review]
LGTM

**Review by danth — COMMENTED** [review]
LGTM in terms of separating the two targets - please consider splitting your changes into separate pull requests in the future so that we can merge them individually.

Perhaps the transparent background for `main` should be enabled automatically when `stylix.opacity.terminal < 1.0`. This is the same thing we do for Helix. It means users with a transparent terminal get a better experience without having to manually enable it, while at the same time, the solid color background is preserved in other cases for compatibility with unthemed terminals.

File: modules/neovim/hm.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{ pkgs, config, lib, ... }:
+
+{
+  options.stylix.targets.neovim = {
+    enable =
+      config.lib.stylix.mkEnableTarget "Neovim" true;
+
+    transparent_bg = {
```
[the remark this answers is not in the document]
Comment (trueNAHO) [reply modules/neovim/hm.nix:8]: Should we also apply this to:

https://github.com/danth/stylix/blob/feb2973dfa8232c07efbd2b48f11a5cfa2276570/modules/nixvim/nixvim.nix#L10

File: modules/neovim/hm.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      main = lib.mkEnableOption "background transparency for the main Neovim window";
+      sign_column = lib.mkEnableOption "background transparency for the Neovim sign column";
```
[the remark this answers is not in the document]
Comment (trueNAHO) [reply modules/neovim/hm.nix:10]: Should we also apply this to:

https://github.com/danth/stylix/blob/feb2973dfa8232c07efbd2b48f11a5cfa2276570/modules/nixvim/nixvim.nix#L12

File: modules/neovim/hm.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (danth) [reply modules/neovim/hm.nix:8]: I didn't notice that this was copied from the NixVim module - ideally yes both modules should use the camelCase format without abbreviations. The NixVim module will need `mkRenamedOptionModule` for backwards compatibility, though renaming those existing options doesn't need to be part of this PR

File: modules/neovim/hm.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (danth) [reply modules/neovim/hm.nix:10]: See this comment https://github.com/danth/stylix/pull/496#discussion_r1704353280

**Comment by danth** [comment]
> In that case, we should also apply this to the [NixVim module](https://github.com/danth/stylix/blob/feb2973dfa8232c07efbd2b48f11a5cfa2276570/modules/nixvim/nixvim.nix).

Perhaps there should be a global setting for "remove the solid background from TUI apps" too, with the default following the behaviour I described above. Basically a global version of the `transparentBackground` option introduced here. (Again, that doesn't need to be done in this PR)

**Review by danth — APPROVED** [review]
Going to merge this now since automatic enablement can be added later, and will need to be implemented for NixVim too which is outside the scope of this PR.

**Pull request merged**